components/CategoryIcon.tsx

```diff
@@ -0,0 +1,46 @@
+import styled from 'styled-components';
+import Image from 'next/image';
+import {
+  CategoryIconProps,
+  StyledIconButtonProps,
+} from '@/types/CategoryIcon.types';
+
+export default function CategoryIcon({
+  category,
+  onChangeFilter,
+  activeFilter,
+}: CategoryIconProps) {
+  const { iconPath, name, colorVarName } = category;
+
+  return (
+    <StyledIconButton
+      $colorVarName={colorVarName}
+      onClick={() => onChangeFilter(name)}
+      $activeFilter={activeFilter}
+      $color={colorVarName}
+      $name={name}
+    >
+      <Image alt={`${name} icon`} src={`${iconPath}`} width={25} height={25} />
+      <span>{name}</span>
+    </StyledIconButton>
+  );
+}
+
+const StyledIconButton = styled.button<StyledIconButtonProps>`
+  background-color: ${({ $activeFilter, $color, $name }) =>
+    $activeFilter === $name ? `var(${$color})` : 'var(--color-neutral-2)'};
```

Comment:
you could move the equality check outside of the styled components and pass down a new prop called `$isActive`. This would reduce the complexity of this component.